test/one_hot_encoding.jl

```diff
@@ -0,0 +1,118 @@
+@testset "one hot encoding" begin
```

Comment:
As well as testing non-unique `categories` (as you noted), could we test
- data having a value not in `categories`
- case where `length(categories) > length(x)` - both (a) to test non-square matrix size both ways, and (b) to test `x` not having all the categories